File: scalang/src/main/scala/com/kapilkoju/autopos/trade/purchase/dto/PurchaseInfo.scala
package com.kapilkoju.autopos.trade.purchase.dto

import java.time.Instant

import com.kapilkoju.autopos.trade.purchase.domain.Purchase

case class PurchaseInfo(id: Long, date: Instant, invoiceNumber: String)

object PurchaseInfo {

  def apply(purchase: Purchase): PurchaseInfo = {
    require(purchase != null)
    PurchaseInfo(purchase.getId, purchase.date, purchase.invoiceNumber)
  }

}

File: src/main/kotlin/com/kapilkoju/autopos/trade/purchase/model/PurchaseInfo.kt
package com.kapilkoju.autopos.trade.purchase.model

import com.kapilkoju.autopos.trade.purchase.domain.Purchase
import java.time.Instant

data class PurchaseInfo(val id: Long, val date: Instant, val invoiceNumber: String?) {
    constructor(purchase: Purchase) : this(purchase.getId()!!, purchase.date, purchase.invoiceNumber)
}

